optuna/_transform.py

```diff
@@ -62,20 +65,26 @@ def __init__(
         search_space: Dict[str, BaseDistribution],
         transform_log: bool = True,
         transform_step: bool = True,
+        transform_0_1: bool = False,
     ) -> None:
         bounds, column_to_encoded_columns, encoded_column_to_column = _transform_search_space(
-            search_space, transform_log, transform_step
+            search_space,
+            transform_log,
+            transform_step,
         )
-
-        self._bounds = bounds
+        self._raw_bounds = bounds
         self._column_to_encoded_columns = column_to_encoded_columns
         self._encoded_column_to_column = encoded_column_to_column
         self._search_space = search_space
         self._transform_log = transform_log
+        self._transform_0_1 = transform_0_1
 
     @property
     def bounds(self) -> numpy.ndarray:
-        return self._bounds
+        if self._transform_0_1:
+            return numpy.array([[0.0, 1.0]] * self._raw_bounds.shape[0])
```

Comment:
How about specifying dtype like the following line?

https://github.com/optuna/optuna/blob/fbd43010efaefcff618dda407f46d076d7889b88/optuna/_transform.py#L185

Reply:
Should we? I think numpy can infer the dtype from 0.0 and 1.0.

Reply:
I see. Let me resolve this.